### PR TITLE
database diagram for a clinic

### PR DESCRIPTION
**Pull request by Ouail-01**
### In this pull request @Bestbynature, @prius2055 and I worked together to implement the SQL needed to build a database based on the diagram below :
Screenshot 2023-06-13 at 11 52 27

- Created a database based on provided diagram
- Added relationships between tables
- Added indexing optimization to database

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+);
+
+-- Create many to many relation :
+
+CREATE TABLE medical_treatments (
+    id INTEGER GENERATED ALWAYS AS IDENTITY,
+    medical_histories_id INTEGER REFERENCES medical_histories(id),
+    treatments_id INTEGER REFERENCES treatments(id),
+    PRIMARY KEY (id)
+);
```
Comment (osamaashraf6) [inline schema_based_on_diagram.sql:59]: - [ ] While you make a great effort 😃 I see that you forgot to make all FOREIGN KEYS have indexes created, as you know the importance of indexing for the optimization of the SQL queries, so please 💯 try to add the index for all the foreign keys to stick to the requirements and also for enhancing the performance 🚀💯

- Hint: 

```snippet
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi @Ouail-01 😊, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
SoCloseAlmostGIF (3)

## Highlights🎉🎉

- [ ] Good using the SQL commands 🚀
- [ ] Good converting the schema into the SQL code👌
- [ ] Good adding the relation many to many.💯

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Bestbynature) [reply schema_based_on_diagram.sql:59]: Thank you so much @osamaashraf6

**Review by binhussen — APPROVED** [review]
Hi @Ouail-01, @prius2055  and @Bestbynature , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Used ER Diagram ✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**